File: algorithms/python/746.py
from functools import lru_cache
class Solution:
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        @lru_cache(maxsize=None)
        def helper(n):
            if n == 0 or n == 1:
                return 0
            else:
                return min(helper(n-2)+cost[n-2], helper(n-1)+cost[n-1])
        return helper(len(cost))

File: algorithms/rust/746.rs
impl Solution {
    #[allow(dead_code)]
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut dp = vec![0; cost.len() + 1];
        for i in 2..=cost.len() {
            dp[i] = (dp[i - 2] + cost[i - 2]).min(dp[i - 1] + cost[i - 1]);
        }
        *dp.last().unwrap()
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
